roll/extensions.py

```diff
@@ -48,6 +50,15 @@ def options(app):
         return request.method == 'OPTIONS'
 
 
+def content_negociation(app):
+
+    @app.listen('dispatch')
+    async def reject_unacceptable_requests(request, payload):
+        accept = request.headers.get('Accept', '*/*')
+        if get_best_match(accept, payload['accepts']) is None:
```

Comment:
I wonder if the return value of `get_best_match` should not be saved in some way?
I guess at some point the value of `Accept` header should be used to define which format to be used as response body?

Reply:
Other thought: should the `route` key be `accepts`, given at the end this is what the view can serve as content type, no?
So `accepts` seems to me the point of view of the request.

Reply:
> I wonder if the return value of get_best_match should not be saved in some way?

It's hard to guess without a use-case, I'd say let's wait for it to appear.

> I guess at some point the value of Accept header should be used to define which format to be used as response body?

Not exactly, it can be a bit different so it's only for the check.

> at the end this is what the view can serve as content type, no?

It can be a bit different, for instance the client asks for `Accept: application/vnd.example.resource+json; version=2` and the server returns `Content-Type: application/vnd.example.resource+json; version=2.1.3`